### ISSUE TITLE
Use ".template" suffix for schematic template files

### ISSUE THREAD

**Issue by devversion** [angular/angular-cli#13378]
Currently in the `cli-hello-world-ivy` integration test for `angular/angular`:

```
Failed to read entry point info from /home/circleci/ng/integration/cli-hello-world-ivy/node_modules/@schematics/angular/workspace/files/package.json with error SyntaxError: Unexpected token < in JSON at position 1121.
```

It looks like this happens because ngcc just tries to read all `package.json` files and **`@schematics/angular`** ships their templated `package.json` file without using the `.template` suffix.

If there is nothing opposed to switching the `package.json` to use `package.json.template`, I think that it would make sense in order to avoid such warnings/errors in general.

Note that this initially caused errors for NGCC, but has been switched to a warning in https://github.com/angular/angular/pull/26539.

**Comment by devversion** [comment]
It's technically not really an issue because nothing is wrong with packaging such templated `package.json` files into an NPM package, but since there already is the possibility to suffix template files with `.template`, I wanted to see if there is a reason why this hasn't been done for the these files.

**Comment by mgechev** [comment]
Makes sense to me. @alan-agius4 pushed it into the queue for discussion this week.
